### PR TITLE
Run Django & Gulp in separate Docker containers

### PR DESCRIPTION
**Pull request by lmorchard**
- Use separate Docker containers for Django and Gulp
- More hacking notes in README
- Stop trying to make django-gulp happen
- Stop trying to notify from gulpfile.js
- .gitignore tweak to ignore VIM swapfiles
- Tweak .travis.yml to use Python 3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,9 +20,31 @@ WIKI: https://wiki.mozilla.org/Idea-Town
...
-* Visit http://0.0.0.0:8000
+* Make sure your shell can see the default Docker machine, i.e.:
+  * `eval "$(docker-machine env default)"`
+
+* Create & set up the Docker containers:
+  * `docker-compose up`
+  * This may take some time.
+
+* To see the IP address of the default Docker machine:
+  * `docker ip default`
```
Comment (jaredhirsch) [inline README.md]: This is great, thanks for including some docs!

For some reason, on my machine, `docker-machine ip default` works, but `docker ip default` doesn't.

File: README.md [hunk 2]
```diff
@@ -20,9 +20,31 @@ WIKI: https://wiki.mozilla.org/Idea-Town
...
+
+* To see the IP address of the default Docker machine:
+  * `docker ip default` 
+
+* To visit the Django server:
+  * `open http://$(docker-machine ip default):8000/`
+  * (or, whatever IP was reported by `docker ip` and port 8000)
+
+* To shell into one of the containers, e.g. to run Django commands:
+  * `docker exec -t -i ideatown_server_1 bash`
```
Comment (jaredhirsch) [inline README.md:45]: Hmm, I think this should be `mozillaideatown_server_1`--or at least, that's what I see when I `docker ps` on my machine

**Comment by jaredhirsch** [comment]
@lmorchard The static server seems to be broken for me. Have you run into this same error?

```
client_build_1 | /app/node_modules/gulp-sass/node_modules/node-sass/lib/extensions.js:148
client_build_1 |     throw new Error(['`libsass` bindings not found in ', binaryPath, '. Try re
client_build_1 |           ^
client_build_1 | Error: `libsass` bindings not found in /app/node_modules/gulp-sass/node_modules/node-sass/vendor/linux-x64-14/binding.node. Try reinstalling `node-sass`?
client_build_1 |     at Object.sass.getBinaryPath (/app/node_modules/gulp-sass/node_modules/node-sass/lib/extensions.js:148:11)
client_build_1 |     at Object.<anonymous> (/app/node_modules/gulp-sass/node_modules/node-sass/lib/index.js:16:36)
client_build_1 |     at Module._compile (module.js:460:26)
client_build_1 |     at Object.Module._extensions..js (module.js:478:10)
client_build_1 |     at Module.load (module.js:355:32)
client_build_1 |     at Function.Module._load (module.js:310:12)
client_build_1 |     at Module.require (module.js:365:17)
client_build_1 |     at require (module.js:384:17)
client_build_1 |     at Object.<anonymous> (/app/node_modules/gulp-sass/index.js:163:21)
client_build_1 |     at Module._compile (module.js:460:26)
```

File: README.md (hunk 1 above)
Comment (pdehaan) [inline README.md]: What version of Docker are y'all running?

```
✗ docker version

Client:
 Version:      1.8.0
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   0d03096
 Built:        Tue Aug 11 17:17:40 UTC 2015
 OS/Arch:      darwin/amd64
```

Comment (lmorchard) [inline README.md]: Yeah, that should be `docker-machine ip default`. And this is the version I'm using, freshly installed from the Docker site: 

```
➜  idea-town git:(moar-docker) ✗ docker version
Client:
 Version:      1.8.1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   d12ea79
 Built:        Thu Aug 13 02:49:29 UTC 2015
 OS/Arch:      darwin/amd64

Server:
 Version:      1.8.1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   d12ea79
 Built:        Thu Aug 13 02:49:29 UTC 2015
 OS/Arch:      linux/amd64
```

File: README.md (hunk 2 above)
Comment (lmorchard) [inline README.md:45]: Weird... my `docker ps` looks like this:

```
➜  idea-town git:(moar-docker) ✗ docker ps
CONTAINER ID        IMAGE                   COMMAND                  CREATED             STATUS              PORTS                    NAMES
98df950da56d        ideatown_server         "./bin/run-dev.sh"       2 hours ago         Up About an hour    0.0.0.0:8000->8000/tcp   ideatown_server_1
cb5a48c38e25        ideatown_client_build   "./bin/run-client-bui"   2 hours ago         Up About an hour    0.0.0.0:9988->9988/tcp   ideatown_client_build_1
ea2cc2fb50f7        postgres:9.3            "/docker-entrypoint.s"   3 hours ago         Up About an hour    5432/tcp                 ideatown_db_1
```

**Comment by lmorchard** [comment]
If you have a `node_modules` in your project dir, I think you have to delete it. The gulp container installs dependencies elsewhere in the container (ie. `/home/webdev/node_modules`). But, it will use the one in the project directory instead if found. And, if that `node_modules` was installed under OS X, it will be missing the `libsass` binaries for Linux.

Ran into this a few times, but thought I'd had it sorted out. Might be nice to find a way to have both OS X and Linux versions of the `libsass` binaries installed

File: README.md (hunk 2 above)
Comment (jaredhirsch) [inline README.md:45]: Whoa, that's odd. I'll try creating a fresh repo and see which names are listed.

**Comment by lmorchard** [comment]
TL;DR: There we go, one more tweak to make Gulp & Django co-exist.

Turns out Django wants to do some static asset management itself. It collects assets from various apps like `admin` & etc into `static/` for production deployment. But, it doesn't serve up assets from `static/` in _development_. 

So, just dumping things built by Gulp into `static/` wasn't working. But, dumping them in `client-build/` and adding that dir to [STATICFILES_DIRS](https://docs.djangoproject.com/en/dev/ref/settings/#std:setting-STATICFILES_DIRS) makes Django serve them from the `/static/` URL, along with other apps like `admin`

File: README.md (hunk 2 above)
Comment (jaredhirsch) [inline README.md:45]: Oh, I see what happened here.

Docker uses the current directory to generate the first part of the name; my clone lives in a directory named `mozilla-idea-town`, and the resulting name is `mozillaideatown_*`.

I'm ok with not documenting this quirk; the names can always be obtained from `docker ps`, and the last part of the name is always the expected thing.

**Comment by jaredhirsch** [comment]
Landing this & we'll take another pass at the static VM magic soon
